Mathlib/Analysis/NormedSpace/UnitizationL1.lean

```diff
@@ -0,0 +1,135 @@
+-/
+
+variable (𝕜 A : Type*) [NormedField 𝕜] [NonUnitalNormedRing A]
+variable [NormedSpace 𝕜 A] [IsScalarTower 𝕜 A A] [SMulCommClass 𝕜 A A]
+
+namespace WithLp
+
+open Unitization
+
+/-- The natural map between `Unitization 𝕜 A` and `𝕜 × A`, transferred to their `WithLp 1`
+synonyms. -/
+noncomputable def unitization_addEquiv_prod : WithLp 1 (Unitization 𝕜 A) ≃+ WithLp 1 (𝕜 × A) :=
+  (WithLp.linearEquiv 1 𝕜 (Unitization 𝕜 A)).toAddEquiv.trans <|
+    (addEquiv 𝕜 A).trans (WithLp.linearEquiv 1 𝕜 (𝕜 × A)).symm.toAddEquiv
+
+noncomputable instance instUnitizationNormedAddCommGroup :
+    NormedAddCommGroup (WithLp 1 (Unitization 𝕜 A)) :=
+  NormedAddCommGroup.induced (WithLp 1 (Unitization 𝕜 A)) (WithLp 1 (𝕜 × A))
+    (unitization_addEquiv_prod 𝕜 A) (AddEquiv.injective _)
+
+/-- Bundle `WithLp.unitization_addEquiv_prod` as a `UniformEquiv`. -/
+noncomputable def uniformEquiv_unitization_addEquiv_prod :
+    WithLp 1 (Unitization 𝕜 A) ≃ᵤ WithLp 1 (𝕜 × A) :=
+  { unitization_addEquiv_prod 𝕜 A with
+    uniformContinuous_invFun := uniformContinuous_comap' uniformContinuous_id
+    uniformContinuous_toFun := uniformContinuous_iff.mpr le_rfl }
+
+instance instCompleteSpace [CompleteSpace 𝕜] [CompleteSpace A] :
+    CompleteSpace (WithLp 1 (Unitization 𝕜 A)) :=
+  completeSpace_congr (uniformEquiv_unitization_addEquiv_prod 𝕜 A).uniformEmbedding |>.mpr
+    CompleteSpace.prod
+
+variable {𝕜 A}
+
+open ENNReal in
+lemma unitization_norm_def (x : WithLp 1 (Unitization 𝕜 A)) :
+    ‖x‖ = ‖(WithLp.equiv 1 _ x).fst‖ + ‖(WithLp.equiv 1 _ x).snd‖ := calc
+  ‖x‖ = (‖(WithLp.equiv 1 _ x).fst‖ ^ (1 : ℝ≥0∞).toReal +
+      ‖(WithLp.equiv 1 _ x).snd‖ ^ (1 : ℝ≥0∞).toReal) ^ (1 / (1 : ℝ≥0∞).toReal) :=
+    WithLp.prod_norm_eq_add (by simp : 0 < (1 : ℝ≥0∞).toReal) _
+  _   = ‖(WithLp.equiv 1 _ x).fst‖ + ‖(WithLp.equiv 1 _ x).snd‖ := by simp
+
+lemma unitization_nnnorm_def (x : WithLp 1 (Unitization 𝕜 A)) :
+    ‖x‖₊ = ‖(WithLp.equiv 1 _ x).fst‖₊ + ‖(WithLp.equiv 1 _ x).snd‖₊ :=
+  Subtype.ext <| unitization_norm_def x
+
+lemma unitization_norm_inr (x : A) : ‖(WithLp.equiv 1 (Unitization 𝕜 A)).symm x‖ = ‖x‖ := by
+  simp [unitization_norm_def]
+
+lemma unitization_nnnorm_inr (x : A) : ‖(WithLp.equiv 1 (Unitization 𝕜 A)).symm x‖₊ = ‖x‖₊ := by
+  simp [unitization_nnnorm_def]
+
+lemma unitization_isometry_inr :
+    Isometry (fun x : A ↦ (WithLp.equiv 1 (Unitization 𝕜 A)).symm x) :=
+  AddMonoidHomClass.isometry_of_norm
+    ((WithLp.linearEquiv 1 𝕜 (Unitization 𝕜 A)).symm.comp <| Unitization.inrHom 𝕜 A)
+    unitization_norm_inr
+
+instance instUnitizationRing : Ring (WithLp 1 (Unitization 𝕜 A)) :=
+  inferInstanceAs (Ring (Unitization 𝕜 A))
+
+@[simp]
+lemma unitization_mul (x y : WithLp 1 (Unitization 𝕜 A)) :
+    WithLp.equiv 1 _ (x * y) = (WithLp.equiv 1 _ x) * (WithLp.equiv 1 _ y) :=
+  rfl
+
+instance instSMul {R : Type*} [SMul R 𝕜] [SMul R A] : SMul R (WithLp 1 (Unitization 𝕜 A)) :=
+  inferInstanceAs (SMul R (Unitization 𝕜 A))
```

Comment:
I think this should be found automatically from the general `WithLp` infrastructure?

Reply:
Ah, indeed. I got tricked because there was no `Algebra` instance, but it's obvious why there isn't one of those.